app/src/components/features/rules/RuleBuilder/Header/ActionButtons/CreateRuleButton/index.js

```diff
@@ -131,9 +134,9 @@ const CreateRuleButton = ({
           }
         }
 
-        // It is necessary to redirect to editor after all the states have been updated so
-        // that usePrompt does not block the navigation
-        setRedirectToEditor(true);
+        if (!isRuleEditorModal) {
+          redirectToRuleEditor(navigate, currentlySelectedRuleData.id, "create");
+        }
```

Comment:
move this inside `finally` of this promise since it is not relevant to the logic for saving rule